gammapy/spectrum/results.py

```diff
@@ -0,0 +1,467 @@
+                 flux_points=None):
+        self.spectral_model = spectral_model
+        self.parameters = Bunch(parameters)
+        self.parameter_errors = Bunch(parameter_errors)
+        self.energy_range = EnergyBounds(energy_range).to('TeV')
+        self.fluxes = fluxes
+        self.flux_errors = flux_errors
+        self.flux_points = Table(flux_points)
+
+    @classmethod
+    def from_fitspectrum_json(cls, filename, model=0):
+        """Read FitSpectrum output file
+
+        Several models can be stored in the JSON file, the appropriate model
+        to read can be chosen with the ``model`` parameter.
+        TODO: Should this be a function?
+
+        Parameters
+        ----------
+        filename : str
+            Name of the JSON file to read
+        model : int
+            Model to read from JSON file
+        """
+        import json
+
+        with open(filename) as fh:
+            data = json.load(fh)
+
+        val = data['fit_functions']['fit_functions'][model]
+        e_range = (val['energy_min'], val['energy_max'])
+        energy_range = EnergyBounds(e_range, 'TeV')
+        spectral_model = val['type']
+        parameters = Bunch()
+        parameters_errors = Bunch()
+        for par in val['parameters']:
+            pname = par['name']
+            if pname == 'Index':
+                unit = Unit('')
+                name = 'index'
+            elif pname == 'Norm':
+                unit = val['norm_scale'] * Unit('cm-2 s-1 TeV-1')
+                name = 'norm'
+            elif pname == 'E0':
+                unit = Unit('TeV')
+                name = 'reference'
+            else:
+                raise ValueError('Unkown Parameter: {}'.format(pname))
+            parameters[name] = par['value'] * unit
+            parameters_errors[name] = par['error'] * unit
+
+        fluxes = Bunch()
+        fluxes['1TeV'] = val['flux_at_1'] * Unit('cm-2 s-1')
+        flux_errors = Bunch()
+        flux_errors['1TeV'] = val['flux_at_1_err'] * Unit('cm-2 s-1')
+        flux_points = Table(data=data['flux_graph']['bin_values'], masked=True)
+        flux_points['energy'].unit = 'TeV'
+        flux_points['energy_err_hi'].unit = 'TeV'
+        flux_points['energy_err_lo'].unit = 'TeV'
+        flux_points['flux'].unit = 'cm-2 s-1 TeV-1'
+        flux_points['flux_err_hi'].unit = 'cm-2 s-1 TeV-1'
+        flux_points['flux_err_lo'].unit = 'cm-2 s-1 TeV-1'
+        flux_points['flux_ul'].unit = 'cm-2 s-1 TeV-1'
+
+        return cls(energy_range=energy_range, parameters=parameters,
+                   parameter_errors=parameters_errors,
+                   spectral_model=spectral_model,
+                   fluxes=fluxes, flux_errors=flux_errors,
+                   flux_points=flux_points)
+
+    @classmethod
+    def from_sherpa(cls, covar, filter, model, flux_graph = None):
+        """Create SpectrumFitResult from sherpa objects
+
+        TODO: Should this be a function?
```

Comment:
Classmethod is nice, no? Remove this TODO?

Document parameters? (or leave a TODO to make it easier to remember ...)